pybossa/view/projects.py

```diff
@@ -1262,24 +1263,18 @@ def respond(tmpl):
     guard.stamp(task, get_user_id_or_ip())
 
     # Verify the worker has an unexpired lock on the task. Otherwise, task will fail to submit.
-    locks = _get_locks(project.id, task_id)
-    valid_locks = {key:seconds for key, seconds in locks.items() if seconds > 0}
-    if not valid_locks and mode != 'read_only':
+    timeout, ttl = fetch_lock_for_user(task.project_id, task_id, user_id)
+    remaining_time = float(ttl) - time.time() if ttl else None
+    if (not remaining_time or remaining_time <= 0) and mode != 'read_only':
         flash(gettext("Unable to lock task or task expired. Please cancel and begin a new task."), "error")
     else:
         if mode != 'read_only':
-            # The user already has a valid lock (page reload).
             # Set the original timeout seconds to display in the message.
-            seconds = project.info.get('timeout', DEFAULT_TASK_TIMEOUT)
-            template_args['project']['original_timeout'] = seconds
-
-            # Find the seconds remaining on this task.
-            lock = {key:seconds for key, seconds in locks.items() if key == current_user.id and seconds > 0}
-            if lock:
-                seconds = lock[current_user.id]
-
```

Comment:
I can see that fetch_lock_user_task is now being used to obtain user time remaining on a given task which make sense. Was `lock` variable unused originally and can be safely removed?

Reply:
yes